pkg/release/spec.go

```diff
@@ -0,0 +1,7 @@
+package release
+
+import releasev1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/release/v1alpha1"
+
+type Getter interface {
```

Comment:
My idea here is to provide only this simple interface for object getting, wire its implementation up to all components that need the cached client and finally separate the functions that utilize `Release` objects themselves.